tests/azure_iot_provisioning_client_ut.c

```diff
@@ -0,0 +1,704 @@
+                                                           &ucEndpoint[ 0 ], sizeof( ucEndpoint ),
+                                                           &ucIdScope[ 0 ], sizeof( ucIdScope ),
+                                                           &ucRegistrationId[ 0 ], sizeof( ucRegistrationId ),
+                                                           NULL, ucBuffer, sizeof( ucBuffer ),
+                                                           prvGetUnixTime,
+                                                           &xTransportInterface ),
+                          eAzureIoTProvisioningSuccess );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTProvisioningClient_Init_Success( void ** ppvState )
+{
+    AzureIoTProvisioningClient_t xTestProvisioningClient;
+
+    ( void ) ppvState;
+    will_return( AzureIoTMQTT_Init, eAzureIoTMQTTSuccess );
+
+    assert_int_equal( AzureIoTProvisioningClient_Init( &xTestProvisioningClient,
+                                                       &ucEndpoint[ 0 ], sizeof( ucEndpoint ),
+                                                       &ucIdScope[ 0 ], sizeof( ucIdScope ),
+                                                       &ucRegistrationId[ 0 ], sizeof( ucRegistrationId ),
+                                                       NULL, ucBuffer, sizeof( ucBuffer ),
+                                                       prvGetUnixTime,
+                                                       &xTransportInterface ),
+                      eAzureIoTProvisioningSuccess );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTProvisioningClient_Deinit_Success( void ** ppvState )
+{
+    AzureIoTProvisioningClient_t xTestProvisioningClient;
+
+    ( void ) ppvState;
+
+    will_return( AzureIoTMQTT_Init, eAzureIoTMQTTSuccess );
+    assert_int_equal( AzureIoTProvisioningClient_Init( &xTestProvisioningClient,
+                                                       &ucEndpoint[ 0 ], sizeof( ucEndpoint ),
+                                                       &ucIdScope[ 0 ], sizeof( ucIdScope ),
+                                                       &ucRegistrationId[ 0 ], sizeof( ucRegistrationId ),
+                                                       NULL, ucBuffer, sizeof( ucBuffer ),
+                                                       prvGetUnixTime,
+                                                       &xTransportInterface ),
+                      eAzureIoTProvisioningSuccess );
+
+    AzureIoTProvisioningClient_Deinit( &xTestProvisioningClient );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTProvisioningClient_SymmetricKeySet_Failure( void ** ppvState )
+{
+    AzureIoTProvisioningClient_t xTestProvisioningClient;
+
+    ( void ) ppvState;
+
+    /* Fail AzureIoTProvisioningClient_SetSymmetricKey when null symmetric key is passed */
```

Comment:
Nit: Shouldn't there also be a test when `xTestProvisioningClient==NULL` and also a non-NULL `ucSymmetricKey` but `0` for its length?

Reply:
Similar feedback other places like `AzureIoTProvisioningClient_Register` fail tests where param validation isn't done for handle.